ucey-star/ID8
# Supabase initialization

## Description
This PR introduced Supabase into the codebase, which is a backend-as-a-service (BaaS) platform that provides a real-time database, authentication, and APIs for seamless integration, allowing developers to build scalable applications without managing infrastructure manually. The reason for this decision is to have a quick development cycle and given most of the back-end operations will be crud operations, supabase is a perfect good candidate. Even better supabase integrates seamlessly with Vercel making hosting a breeze!
 
 I've added an API folder to the codebase to hold the Supabase client configuration. Since Supabase is handling most of our back-end operations, I've also introduced a type-checking system to ensure better integration between the front-end and back-end. This allows front-end developers to easily view the expected data structure by simply hovering over the variable storing the fetched results, streamlining the development process.
 
**Note:**  Back-end developers (@ucey-star) will need to run `npm run generate:types` whenever changes are made to the database tables, such as modifying the structure or adding new tables. This ensures the type-checking system stays up-to-date with the latest database schema. 


## Type of Change


## Testing Instructions
To test this PR you will first need to create a .env file in your local and paste in the environment variables which are in the Engineering telegram group. 
From there on the home page click the fetchTestData button and some Test text should show up.


## Screenshots (if applicable)

If your changes include visual components, add screenshots showing the affected pages or elements.
